### PR TITLE
Load "simple" data from XML

### PR DESCRIPTION
**Pull request by tie-rack**
Pivotal story: [91447450](https://www.pivotaltracker.com/story/show/91447450)

Load data from XML uploads into a SQLite database for the following elements:

* ballot
* ballot_line_result
* ballot_response
* contest
* contest_result
* custom_ballot
* election
* election_official
* electoral_district
* locality
* precinct
* precinct-split
* referendum
* source
* state

Code for checking value formats and transforming values for insertion into the database has been generalized from CSV to the shared `data-spec` namespace.

Also:

* Organized test-resources
* Removed `NOT NULL` constraints on non-required columns

### CODE REVIEW

File: src/vip/data_processor/validation/db.clj [hunk 1]
```diff
@@ -13,24 +13,24 @@
       (assoc-in ctx [:errors "Duplicate IDs"] dupes)
       ctx)))
 
-(defn validate-no-duplicated-rows [{:keys [csv-specs] :as ctx}]
-  (reduce dupe-records/validate-no-duplicated-rows-in-table ctx (:csv-specs ctx)))
+(defn validate-no-duplicated-rows [{:keys [data-specs] :as ctx}]
+  (reduce dupe-records/validate-no-duplicated-rows-in-table ctx (:data-specs ctx)))
```
Comment (ericnormand) [inline src/vip/data_processor/validation/db.clj]: Why destructure in params and use keyword to pull it out?

File: src/vip/data_processor/validation/db/duplicate_ids.clj [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-  (some #(-> % :name (= "id")) (:columns csv-spec)))
+(defn has-id? [data-spec]
+  (some #(-> % :name (= "id")) (:columns data-spec)))
 
 (defn tables-with-ids [ctx]
-  (let [csvs-with-id (filter has-id? csv/csv-specs)
-        table-names (map :table csvs-with-id)]
+  (let [table-with-id (filter has-id? data-specs)
+        table-names (map :table table-with-id)]
     (map (:tables ctx) table-names)))
```
Comment (ericnormand) [inline src/vip/data_processor/validation/db/duplicate_ids.clj]: Would this look better as a ->> ?

File: test-resources/xml/full-good-run.xml [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+    <odd_even_both>both</odd_even_both>
+    <non_house_address>
+      <street_name>*</street_name>
+      <city>Rural</city>
+      <state>OH</state>
+      <zip>43321</zip>
+    </non_house_address>
+    <precinct_id>10203</precinct_id>
+  </street_segment>
+</vip_object>
```
Comment (ericnormand) [inline test-resources/xml/full-good-run.xml]: I'm just going to assume this XML is a good test.

**Comment by ericnormand** [comment]
It's a big PR, so I'll ask a question: you're now allowing NULLs in many columns. Where do you handle the nils?

File: src/vip/data_processor/validation/db.clj (hunk 1 above)
Comment (tie-rack) [inline src/vip/data_processor/validation/db.clj]: Idiocy. I'll fix!

File: src/vip/data_processor/validation/db/duplicate_ids.clj (hunk 2 above)
Comment (tie-rack) [inline src/vip/data_processor/validation/db/duplicate_ids.clj]: Good point. Can do.

File: test-resources/xml/full-good-run.xml (hunk 3 above)
Comment (tie-rack) [inline test-resources/xml/full-good-run.xml]: This is "blessed" XML. It should hold up.

**Comment by tie-rack** [comment]
@ericnormand Those columns were erroneously marked as `NOT NULL` before. The nils will be handled in a later step when we're exporting XML. At that point, anything nil is not going to be exported.

**Comment by ericnormand** [comment]
Tests run on my local.

**Comment by tie-rack** [comment]
@ericnormand I addressed those notes.

**Comment by ericnormand** [comment]
LGTM

**Pull request merged**